components/TARGET_PSA/services/attestation/COMPONENT_SPE/psa_attestation_partition.c

```diff
@@ -0,0 +1,259 @@
+/*
+* Copyright (c) 2018-2019 ARM Limited. All rights reserved.
+*
+* SPDX-License-Identifier: Apache-2.0
+*
+* Licensed under the Apache License, Version 2.0 (the License); you may
+* not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// ---------------------------------- Includes ---------------------------------
+#include "psa/service.h"
+#include "psa/client.h"
+
+#define PSA_ATTEST_SECURE 1
+#include "psa_attest_srv_partition.h"
+#include "psa_initial_attestation_api.h"
+#include "psa_attest_inject_key.h"
+#include "psa_inject_attestation_key_impl.h"
+#include "attestation.h"
+#include <string.h>
+#include "mbedtls/entropy.h"
+#include "crypto.h"
+
+#if defined(PLATFORM_C)
+#include "mbedtls/platform.h"
+#else
+#define calloc calloc
+#define free   free
+#endif
+
+int32_t g_caller_id = 0;
+
+static void set_caller_id(psa_msg_t msg)
+{
+    g_caller_id = psa_identity(msg.handle);
+}
+
+// ------------------------- Partition's Main Thread ---------------------------
+
+static void psa_attest_get_token(void)
+{
+    psa_msg_t msg = { 0 };
```

Comment:
This is just blowing code space, isn't it? Or is it needed to handle a silent `psa_get` failure?

Generally a bad idea to initialise variables to dummy data, as it can waste ROM space (particularly if a structure), and it stops the compiler or valgrind's memcheck from spotting paths where you don't fill in the variables with their real data.

Reply:
@kjbracey-arm this is the default way to declare the psa_msg_t type
in most cases it will be zeroed by psa_get

i don't see an issue here

Reply:
> in most cases it will be zeroed by psa_get

I think that neatly articulates the issue. The current code poses a number of questions: Is it an output parameter for `psa_get`, or is it a read-modify-write parameter? Can we affect the behaviour of `psa_get` by having different stuff in the message on entry? Under what circumstances does `psa_get` not write to it?

By not initialising it, that's a clear statement that it's an output-only parameter, and the compiler and/or valgrind can check that (a) `psa_get` doesn't read it and (b) `psa_get` always writes the fields that you rely on.

With the initialisation, the code is just fuzzy about who's supposed to be writing to it, and who reads from it.

And on top of that, this is actually making the code bigger. So that's two negative strikes, and no advantage I can see.

(The output-only API argument would suggest it might be even better to have the API be `psa_msg_t psa_get(psa_signal_t signum)`, but that can be less efficient if the compiler fails to perform return value optimisation, so it's not totally clear-cut).